compiler/compile/OMRCompilation.cpp

```diff
@@ -720,7 +720,22 @@ OMR::Compilation::getOSRInductionOffset(TR::Node *node)
 bool
 OMR::Compilation::requiresPreOSRPoint(TR::Node *node)
    {
-   return node->getOpCodeValue() != TR::monent;
+   // When there is no OSR induction offset, a pre OSR point is required
+   // This currently only covers async checks as they save the state when the check is performed
+   if (getOSRInductionOffset(node) == 0)
```

Comment:
I see, so the asyncheck is somewhat special because it's not actually in the bytecodes of the method. 

monent and calls are both in the method bytecodes and behave in the ways we would expect : we have an OSR point after we return from the monent and for the call is both before and after to account for the inlining possibility and the place of the actual OSR respectively.